File: solutions/1929 - Concatenation of Array/array_concatenation.cpp
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        // Time Complexity: O(N) to build the answer array
        // Space Complexity: O(N) for answer array
        vector<int> newArr(nums.size() * 2);
        for (int i = 0; i < nums.size(); i++)
        {
            newArr[i] = nums[i];
        }
        for (int i = nums.size(); i < newArr.size(); i++)
        {
            newArr[i] = nums[i - nums.size()];
        }
        return newArr;
    }
};

int main() {
    cout << "Enter the elements of the array: " << endl;

    string line;
    getline(cin, line);

    stringstream ss(line);
    vector<int> nums;
    int x;

    while (ss >> x) {
        nums.push_back(x);
    }

    Solution solution;
    vector<int> result = solution.getConcatenation(nums);

    cout << "Concatenated array:" << endl;
    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    cout << endl;

    return 0;
}

File: solutions/1929 - Concatenation of Array/array_concatenation.java
import java.util.*;
class Solution {
    public int[] getConcatenation(int[] nums) {
        // Time Complexity: O(N) to build the answer array.
        // Space Complexity: O(N) for answer array.
        int n = nums.length;
        int[] answer = new int[n * 2];
        for (int i = 0; i < n; i++) {
            answer[i] = answer[i + n] = nums[i];
        }
        return answer;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter array elements: ");
        String line = sc.nextLine();

        String[] parts = line.trim().split("\\s+");
        int[] nums = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }

        Solution solution = new Solution();
        int[] result = solution.getConcatenation(nums);

        System.out.print("Result: ");
        for (int val : result) {
            System.out.print(val + " ");
        }
        System.out.println();

        sc.close();
    }
}